clients/src/main/java/org/apache/kafka/clients/ClientDnsLookup.java

```diff
@@ -22,7 +22,8 @@
 
     DEFAULT("default"),
     USE_ALL_DNS_IPS("use_all_dns_ips"),
-    RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY("resolve_canonical_bootstrap_servers_only");
+    RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY("resolve_canonical_bootstrap_servers_only"),
+    USE_FIRST_DNS_IP("use_first_dns_ip");
```

Comment:
We said this would be deprecated for removal in 3.0. Let's mention that in the documentation and add a warning if this is selected.

Reply:
Thinking about it, maybe we should not add this at all. We could deprecate `default` and switch the default to `use_all_dns_ips`. Thoughts @badaiaqrandista and @rajinisivaram?

Reply:
I think that is a better idea, instead of introducing new value that would be removed soon.

`ClientDnsLookup.DEFAULT` is used in few places in core (server):

https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaServer.scala#L520
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ReplicaFetcherBlockingSend.scala#L92
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/controller/ControllerChannelManager.scala#L156
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala#L82

And a couple of tools:
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/tools/ReplicaVerificationTool.scala#L482
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/BrokerApiVersionsCommand.scala#L299

It is also used literally in a lot of test cases under clients.

I did not want to touch too many code in the first go. Should I change them all in this KIP or leave them until we remove `ClientDnsLookup.DEFAULT` in 3.0 ?

Reply:
@badaiaqrandista I think you should change it in the non test code in this PR. For the test code, we can do it in a separate PR.

Reply:
Ok. I have changed core code to use `ClientDnsLookup.USE_ALL_DNS_IPS` and leaving all clients tests except `ClientUtilsTest`.